test/skimNtuple_HHbtag.cpp

```diff
@@ -2187,48 +2247,17 @@ int main (int argc, char** argv)
 			  }
 			}
 
-		  if (pairType == 0) { //mutau
-			if(PERIOD=="2018") {
-			  MET_region = ((tlv_firstLepton.Pt() < 25. and tlv_secondLepton.Pt() < 32.) or
-							(tlv_firstLepton.Pt() < 21. and tlv_secondLepton.Pt() < tau_thresh));
-			  SingleTau_region = tlv_firstLepton.Pt() < 21. and tlv_secondLepton.Pt() >= tau_thresh;
-			}
-			else if(PERIOD=="2017") {
-			  MET_region = ((tlv_firstLepton.Pt() < 28. and tlv_secondLepton.Pt() < 32.) or
-							(tlv_firstLepton.Pt() < 21. and tlv_secondLepton.Pt() < tau_thresh));
-			  SingleTau_region = tlv_firstLepton.Pt() < 21. and tlv_secondLepton.Pt() >= tau_thresh;
-			}
-			else if (PERIOD=="2016preVFP" or PERIOD=="2016postVFP") {
-			  MET_region = ((tlv_firstLepton.Pt() < 25. and tlv_secondLepton.Pt() < 25.) or
-							(tlv_firstLepton.Pt() < 20. and tlv_secondLepton.Pt() < tau_thresh));
-			  SingleTau_region = tlv_firstLepton.Pt() < 20. and tlv_secondLepton.Pt() >= tau_thresh;
-			}
-		  }
+		  trgRegions = assignTriggerRegions(pairType, PERIOD,
+											tlv_firstLepton.Pt(), tlv_secondLepton.Pt(),
+											fabs(tlv_firstLepton.Eta()), fabs(tlv_secondLepton.Eta()),
+											eleEtaMax, muEtaMax, tau_thresh);
 
-		  else if (pairType == 1) { //etau
-			if(PERIOD=="2018" or PERIOD=="2017") {
-			  MET_region = ((tlv_firstLepton.Pt() < 33. and tlv_secondLepton.Pt() < 35.) or
-							(tlv_firstLepton.Pt() < 25. and tlv_secondLepton.Pt() < tau_thresh));
-			  SingleTau_region = tlv_firstLepton.Pt() < 25. and tlv_secondLepton.Pt() >= tau_thresh;
-			}
-			else if (PERIOD=="2016preVFP" or PERIOD=="2016postVFP") {
-			  MET_region = tlv_firstLepton.Pt() < 25. and tlv_secondLepton.Pt() < tau_thresh;
-			  SingleTau_region = tlv_firstLepton.Pt() < 25. and tlv_secondLepton.Pt() >= tau_thresh;
-			}
-		  }
-
-		  else if (pairType == 2) { //tautau
-			MET_region       = ((tlv_firstLepton.Pt() < 40 and tlv_secondLepton.Pt() < tau_thresh) or
-								(tlv_firstLepton.Pt() < tau_thresh and tlv_secondLepton.Pt() < 40));
-			SingleTau_region = ((tlv_firstLepton.Pt() < 40 and tlv_secondLepton.Pt() >= tau_thresh) or
-								(tlv_firstLepton.Pt() >= tau_thresh and tlv_secondLepton.Pt() < 40));
-		  }
-
-		  bool metAccept = passMETTrgNoThresh and !passTrg and MET_region; //!passTrg should be redundant wrt to the region cut
-		  bool singletauAccept = passSingleTau and !passTrg and SingleTau_region; //!passTrg should be redundant wrt to the region cut
+		  bool legacyAccept    = passTrg            and trgRegions["legacy"];
```

Comment:
wouldn't this cause some difference between data & MC? for MC, `triggeraccept` a bit below will only use `passTrg` while for data it'll use this `legacyAccept` (from redefining `passTrg` on line 2259)

Reply:
Nice bug catch. I was supposed to replace ```passTrg``` by ```legacyAccept``` in two places.